gap/oper.gi

```diff
@@ -377,6 +377,237 @@ InstallMethod(DigraphClosure,
 [IsImmutableDigraph, IsPosInt],
 {D, k} -> MakeImmutable(DigraphClosure(DigraphMutableCopy(D), k)));
 
+InstallMethod(DigraphContractEdge,
+"for a mutable digraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(D, u, v)
+  local w, neighbours, transformations_to_edge, transformations_to_old_edges,
+  old_edge, new_edge, neighbour, t, copy, vertices;
+
+  if not IsDigraphEdge(D, u, v) then  # Check if [u, v] is an edge in digraph D
+    ErrorNoReturn("u, v is not an edge of D");
+  fi;
+
+  if IsMultiDigraph(D) then  # multi digraphs cannot have their edges contracted
```

Comment:
These checks should be combined into a single `if A then Error; elif B then Error;...` if you see what I mean